### DISCUSSION TITLE
Internal Link Preview in the PDF Preview?

### DISCUSSION THREAD

**Discussion by wayoflife** [James-Yu/LaTeX-Workshop#3958] [Q&A] [answered]
A very useful feature for reading scientific papers is a small popup with the linked content when hovering over such a link. The functionality is found in many pdf readers and has long since been requested and built for pdf.js, see [this issue](https://github.com/mozilla/pdf.js/issues/5835).

One implementation that does this for FireFox and Chrome can be found [here](https://github.com/belinghy/PDFRefPreview).

My question is, is this functionality already part of the pdf preview? If so, how can this be done?

If not (I'm not very familiar with the internals of vs code or writing any kind of extensions for it) is it reasonable to include this feature given the above implementation?

**Comment by James-Yu** [comment, marked as answer]
I quite agree with the comments in https://github.com/mozilla/pdf.js/issues/5835#issuecomment-462041701, plus, LW patches pdf.js in a quite hacky way, which is not very maintainable. Adding a complex feature (though may not seem so, but just e.g. think of how to compute the width of the popup) like this will make the codebase much more un-maintainable. This drives me less interested in developing this feature.

There were a previous issue discussion the exact same feature.

Though I still welcome PR contributions, as long as the code is obviously easy-to-maintain. This may have a quite high bar though.
